### PR TITLE
tpm2_nvread: Add "out-file" option (-f)

### PR DESCRIPTION
**Pull request by harisokanovic**
Add "out-file" option (-f) to save data to a user-specified file in
addition to hexdump()-ing to screen.

Signed-off-by: Haris Okanovic <haris.okanovic@ni.com>

### CODE REVIEW

File: tools/tpm2_nvread.c [hunk 1]
```diff
@@ -133,7 +134,7 @@ static bool nv_read(TSS2_SYS_CONTEXT *sapi_context) {
         return false;
     }
 
-
+    FILE *outputFp = NULL;
```
Comment (williamcroberts) [inline tools/tpm2_nvread.c]: nO cAmEl cAsing

Read: misc/coding_standard_c.txt

Also, variables should be declared at use

File: tools/tpm2_nvread.c [hunk 2]
```diff
@@ -240,6 +262,7 @@ bool tpm2_tool_onstart(tpm2_options **opts) {
     const struct option topts[] = {
         { "index"       , required_argument, NULL, 'x' },
         { "authHandle"  , required_argument, NULL, 'a' },
+        { "out-file"    , required_argument, NULL, 'f' },
```
Comment (williamcroberts) [inline tools/tpm2_nvread.c]: output not out-file

File: tools/tpm2_nvread.c [hunk 3]
```diff
@@ -162,10 +163,28 @@ static bool nv_read(TSS2_SYS_CONTEXT *sapi_context) {
...
+
+    /* dump data_buffer to output file, if specified */
+    if (ctx.output_file) {
+        outputFp = fopen(ctx.output_file, "w+");
+        if (!outputFp) {
+            LOG_ERR("Failed to open output file");
+            goto out;
+        }
+
+        if (fwrite(data_buffer, data_offset, 1, outputFp) != 1) {
```
Comment (williamcroberts) [inline tools/tpm2_nvread.c]: we have file binary output routines in files.h, use em.

File: tools/tpm2_nvread.c [hunk 4]
```diff
@@ -162,10 +163,28 @@ static bool nv_read(TSS2_SYS_CONTEXT *sapi_context) {
         data_offset += nv_data.t.size;
     }
 
+    /* dump data_buffer to stdout */
     hexdump(data_buffer, data_offset);
```
Comment (williamcroberts) [inline tools/tpm2_nvread.c]: I guess this is fine, if you want just file only one could use `-Q -o out.dat`

**Comment by harisokanovic** [comment]
Hi William, I updated my branch with your feedback and added documentation (commit f925fdb9). Let me know if there are any other issues.

**Pull request merged**